test/e2e/nodepool_list.go

```diff
@@ -0,0 +1,397 @@
+				map[string]interface{}{
+					"clusterName":     clusterName,
+					"persistTagValue": false,
+				},
+				45*time.Minute,
+			)
+			Expect(err).NotTo(HaveOccurred())
+
+			npClient := tc.Get20240610ClientFactoryOrDie(ctx).NewNodePoolsClient()
+			resourcesClient := tc.GetARMResourcesClientFactoryOrDie(ctx).NewClient()
+
+			// ========================================
+			// PHASE 1: Verify empty nodepool listings
+			// ========================================
+
+			By("listing nodepools by cluster - should be empty")
+			clusterPager := npClient.NewListByParentPager(*resourceGroup.Name, clusterName, nil)
+			nodePoolCount := 0
+			for clusterPager.More() {
+				nodePoolList, err := clusterPager.NextPage(ctx)
+				Expect(err).To(BeNil())
+				nodePoolCount += len(nodePoolList.Value)
+			}
+			Expect(nodePoolCount).To(Equal(0), "Expected no nodepools in cluster without nodepools")
+
+			By("listing nodepools by resource group - should be empty")
+			resourcesGroupPager := resourcesClient.NewListByResourceGroupPager(*resourceGroup.Name, &armresources.ClientListByResourceGroupOptions{
+				Filter: ptr.To("resourceType eq 'Microsoft.RedHatOpenShift/hcpOpenShiftClusters/nodePools'"),
+			})
+			for resourcesGroupPager.More() {
+				nodePoolList, err := resourcesGroupPager.NextPage(ctx)
+				Expect(err).To(BeNil())
+				nodePoolCount += len(nodePoolList.Value)
+			}
+			Expect(nodePoolCount).To(Equal(0), "Expected no nodepools in RG")
+
+			By("listing nodepools by subscription - should find none from created cluster")
+			subscriptionPager := resourcesClient.NewListPager(&armresources.ClientListOptions{
+				Filter: ptr.To("resourceType eq 'Microsoft.RedHatOpenShift/hcpOpenShiftClusters/nodePools'"),
+			})
+			foundOurClusterNodePool := false
+			for subscriptionPager.More() {
+				nodePoolList, err := subscriptionPager.NextPage(ctx)
+				Expect(err).To(BeNil())
+				for _, np := range nodePoolList.Value {
+					Expect(*np.ID).ToNot(BeEmpty())
+					if np.Name != nil && strings.HasPrefix(*np.Name, clusterName+"/") {
+						foundOurClusterNodePool = true
+						break
+					}
+				}
+				if foundOurClusterNodePool {
+					break
+				}
+			}
+			Expect(foundOurClusterNodePool).To(BeFalse(), "Expected no nodepools from created cluster in subscription listing")
+
+			// ========================================
+			// PHASE 2: Create a nodepool
+			// ========================================
+
+			By("creating nodepool with 2 replicas")
+			_, err = framework.CreateBicepTemplateAndWait(ctx,
+				tc.GetARMResourcesClientFactoryOrDie(ctx).NewDeploymentsClient(),
+				*resourceGroup.Name,
+				"nodepool-deployment",
+				framework.Must(TestArtifactsFS.ReadFile("test-artifacts/generated-test-artifacts/modules/nodepool.json")),
+				map[string]interface{}{
+					"clusterName":  clusterName,
+					"nodePoolName": nodePoolName1,
+					"replicas":     2,
+				},
+				45*time.Minute,
```

Comment:
12 minutes is probably long enough to wait for a node pool to create, at least based on the installation timings I have recently done.